src/StableHashTraits.jl

```diff
@@ -24,17 +58,18 @@ write(io, x) = Base.write(io, x)
 function stable_hash_helper(x, hash, context, ::UseWrite)
     io = IOBuffer()
     write(io, x, context)
-    return hash(take!(io))
+    update!(hash, take!(io))
+    return hash
 end
 
 struct UseIterate end
 function stable_hash_helper(x, hash, context, ::UseIterate)
-    result = hash(UInt8[])
     for el in x
-        val = stable_hash_helper(el, hash, context, hash_method(el, context))
-        result = hash(copy(reinterpret(UInt8, [val])), result)
+        val = stable_hash_helper(el, similar_hasher(hash), context,
+                                 hash_method(el, context))
+        update!(hash, copy(reinterpret(UInt8, vcat(digest!(val)))))
```

Comment:
I'm not understanding the `vcat` here.

Reply:
It ensures the object is an array; `vcat(1) == [1]`; `vcat([1]) == [1]`